qldbdriver/transaction.go

```diff
@@ -107,20 +107,26 @@ type transactionExecutor struct {
 }
 
 // Execute a statement with any parameters within this transaction.
-func (executor *transactionExecutor) Execute(statement string, parameters ...interface{}) (*Result, error) {
+func (executor *transactionExecutor) Execute(statement string, parameters ...interface{}) (Result, error) {
 	return executor.txn.execute(executor.ctx, statement, parameters...)
 }
 
-// Buffer a Result into a BufferedResult to use outside the context of this transaction.
-func (executor *transactionExecutor) BufferResult(result *Result) (*BufferedResult, error) {
+// Buffer a result into a bufferedResult to use outside the context of this transaction.
+func (executor *transactionExecutor) BufferResult(res Result) (*bufferedResult, error) {
 	bufferedResults := make([][]byte, 0)
-	for result.Next(executor) {
-		bufferedResults = append(bufferedResults, result.GetCurrentData())
+	for res.Next(executor) {
+		bufferedResults = append(bufferedResults, res.GetCurrentData())
 	}
-	if result.Err() != nil {
-		return nil, result.Err()
+	if res.Err() != nil {
+		return nil, res.Err()
 	}
-	return &BufferedResult{bufferedResults, 0, nil, result.metrics}, nil
+
+	var metrics *metrics = nil
+	if r, ok := res.(*result); ok {
```

Comment:
Why is this block of code needed?

Reply:
Because the passed in `res` is of interface type `Result` (not concrete type `result`) so this is needed to get the metrics from the underlying `result` object.

Reply:
I see, but this is still type checking and casting when in this case we don't need to do this.

We can just change bufferedResult to take in the metrics individually after we get them from Result and change its functionality to return them individually.

Reply:
Addressed.